draft-ietf-tls-esni.md

```diff
@@ -994,6 +966,72 @@ The "ech_is_inner" is expected to have an empty payload. If the payload is
 non-empty (i.e., the length of the "extension_data" field is non-zero) then the
 backend server MUST abort the handshake with an "illegal_parameter" alert.
 
+## Handling HelloRetryRequest {#server-hrr}
+
+The HelloRetryRequest constructed by the backend server is called the
+HelloRetryRequestInner message; and the HelloRetryRequest constructed by the
+client-facing server is called the HelloRetryRequestOuter message.
+
+If the backend server cannot complete the handshake with the parameters provided
+in the ClientHelloInner, then it constructs a HelloRetryRequestInner message in
+the usual way and forwards it to the client-facing server. The client-facing
+server encrypts this message using a key derived from the HPKE context, as
+described below. It then constructs HelloRetryRequestOuter message with an
+"encrypted_client_hello" extension containing the ciphertext as its payload. It
+then continues the handshake by sending this message to the client.
+
+The payload of the "encrypted_client_hello" extension is constructed as follows.
+The client-facing server begins by computing `hrr_inner_key` and
+`hrr_inner_nonce` as described in {{client-hrr-accepted-ech}}. The extension
+payload is then computed as
+
+~~~
+    encrypted_hrr_inner = Seal(hrr_inner_key,
+                               hrr_inner_nonce,
+                               "", // Empty AAD
+                               HelloRetryRequestInner)
+~~~
```

Comment:
This text is also duplicative.

Reply:
What does it duplicate? Here we specify the encryption operation; the decryption operation is specified above, in {{client-hrr-accpted-ech}}.

Reply:
I see what you're saying. I don't think this is that helpful a style, but I'll let it go for now.

Reply:
Ack.